### ISSUE TITLE
Accessibility md-select aria-owns

### ISSUE THREAD

**Issue by Lindsor** [angular/material2#7023]
Bug:

aria-owns in the DOM even though the elements it points to are not

What is the expected behavior?

aria-owns should not point to elements not in the DOM, if the elements are not in the DOM then aria-owns should be removed as well

What is the current behavior?

aria-owns shows up in the DOM when those elements are not in the DOM

What are the steps to reproduce?

http://embed.plnkr.co/cLXRzvn2uWQ5y5YmiU8L/
inspect select, notice non-empty `aria-owns`

What is the use-case or motivation for changing an existing behavior?

Currently throws accessibility issues

Which versions of Angular, Material, OS, TypeScript, browsers are affected?

"@angular/material": "2.0.0-beta.8"

Is there anything else we should know?

**Comment by crisbeto** [comment]
It does make sense to clear the attribute when the panel is hidden, however it looks like it causes a regression in the user experience. Previously NVDA would read out something like "Option text, selected" when selecting, but it doesn't do it anymore if the attribute is cleared. Also [looking at the documentation](https://www.w3.org/TR/wai-aria-1.1/#aria-owns), it seems like pointing to unexisting elements isn't disallowed explicitly.

**Comment by Lindsor** [comment]
Hmm yes reading the spec theres no specific mention of not allowing it. But if you read this part it implies it.

```
The value of the aria-owns attribute is a space-separated list of IDREFS that reference one or more elements in the document by ID.
```

It says "one or more elements **IN** the document by ID.

I'm ok with leaving it if thats what you think though.
Our business has flagged it as an accessibility bug we need to fix.

Let me know what you think, I'll create a PR if you think we should implement

**Comment by drewterry** [comment]
This causes issues with [axe-core](https://github.com/dequelabs/axe-core) when testing for aria roles and required children (listbox requires options).  If the aria-owns tag is removed when the listbox is closed, the test fails even though the tags are added when the listbox is opened.

**Comment by lozanol** [comment]
This is also an issue when using the [Chrome accessibility tools](https://github.com/GoogleChrome/accessibility-developer-tools)
